engine/access/rpc/backend/node_communicator.go

```diff
@@ -39,8 +49,8 @@ func NewNodeCommunicator(circuitBreakerEnabled bool) *NodeCommunicator {
 // If the maximum failed request count is reached, it returns the accumulated errors.
 func (b *NodeCommunicator) CallAvailableNode(
 	nodes flow.IdentityList,
-	call NodeAction,
-	shouldTerminateOnError ErrorTerminator,
+	call func(id *flow.Identity) error,
+	shouldTerminateOnError func(node *flow.Identity, err error) bool,
```

Comment:
please use the original types. They help document the callbacks

Reply:
this was added to avoid import cycle due to mock -> backend -> mock dependency.

Reply:
If i add them back I'll have to amend makefile mockgen which will differ from conventional way.

Reply:
ok. then please move the comments from the function types into the `CallAvailableNode` godocs so devs can easily see the usage details. You can remove those  function types if they're unused.